### PR TITLE
WIP: Quick spike on storing dependencies for resources

### PR DESCRIPTION
**Pull request by davidfowl**
A quick spike on how we would store dependencies in the resource model

### CODE REVIEW

File: src/Aspire.Hosting/Extensions/ResourceExtensions.cs [hunk 1]
```diff
@@ -118,6 +118,16 @@ public static bool TryGetContainerImageName(this IResource resource, [NotNullWhe
         return false;
     }
 
+    /// <summary>
+    /// 
+    /// </summary>
+    /// <param name="resource"></param>
+    /// <returns></returns>
+    public static IEnumerable<IResource> GetDependencies(this IResource resource)
```
Comment (mitchdenny) [inline src/Aspire.Hosting/Extensions/ResourceExtensions.cs:126]: Should this be transitive?

File: src/Aspire.Hosting/Publishing/ManifestPublisher.cs [hunk 2]
```diff
@@ -136,6 +136,22 @@ private static void WriteServiceDiscoveryEnvironmentVariables(IResource resource
             }
         }
     }
+    internal static void WriteDependencies(IResource resource, ManifestPublishingContext context)
```
Comment (mitchdenny) [inline src/Aspire.Hosting/Publishing/ManifestPublisher.cs:139]: Need to consider how to reuse this logic for custom publishing callbacks outside this assembly.

File: src/Aspire.Hosting/Extensions/ResourceExtensions.cs (hunk 1 above)
Comment (davidfowl) [reply src/Aspire.Hosting/Extensions/ResourceExtensions.cs:126]: No, I think the graph walk should be another API. For e.g. we wouldn't want to write the closure to the manifest, but we would walk it when walking the graph to do something in process.

**Comment by ellismg** [comment]
> So we are writing the dependencies to the manifest. But what the scenarios where that information is going to be useful, and do we need to sub-class the dependency annotation in some instances to capture additional information about why the the dependency exists.

FWIW - `azd` could use this dependency information during `deploy` to schedule the order in which services are deployed (today we just deploy them in alphabetical order).  I recall in a much earlier iteration of the manifest we did have this `dependencies` block and I had always thought I might use it in this manner.

Wondering: Will you ensure that the dependency graph is acyclic or would consumers of the manifest need to take care to ensure that if the use this information there might be cycles?

**Comment by bwateratmsft** [comment]
I'm working on a change on top of this branch that will reenable proxyless containers

**Pull request closed without merging**